src/server_manager/web_app/ui_components/app-root.js

```diff
@@ -561,15 +561,34 @@ export class AppRoot extends mixinBehaviors
    * @param {string} language
    * @param {string} direction
    */
-  setLanguage(language, direction) {
+  async setLanguage(language, direction) {
+    const out = new Promise((resolve, reject) => {
+      // loadResources uses events and continuation instead of Promises.  In order to make this
+      // function easier to use, we wrap the language-changing logic in event handlers which
+      // resolve or reject the Promise.  Note that they need to clean up whichever event handler
+      // didn't fire so we don't leak it, which could cause future language changes to not work
+      // properly.
+      let successHandler, failureHandler;
+      successHandler = () => {
+        this.removeEventListener('app-localize-resources-error', failureHandler);
+        const alignDir = direction === 'ltr' ? 'left' : 'right';
+        this.$.appDrawer.align = alignDir;
+        this.$.sideBar.align = alignDir;
+        this.language = language;
+        resolve();
+      };
+      failureHandler = (err) => {
```

Comment:
The input here is an event, not an error, but I think it will be empty: https://github.com/PolymerElements/app-localize-behavior/blob/e796bfa6876498608ab2d0d63c89bffece80c0da/app-localize-behavior.js#L327

Reply:
Done in 9d11943e51379020fda0b9357457e185e844dfc3